src/Razor/src/rzls/NamedPipeBasedRazorProjectInfoDriver.cs

```diff
@@ -46,15 +46,34 @@ protected override void OnDispose()
     private async Task ReadFromStreamAsync(CancellationToken cancellationToken)
     {
         Logger.LogTrace($"Starting read from named pipe.");
+        var failedActionReads = 0;
 
         while (
             _namedPipe is { IsConnected: true } &&
             !cancellationToken.IsCancellationRequested)
         {
+            ProjectInfoAction? projectInfoAction;
+            try
+            {
+                projectInfoAction = _namedPipe.ReadProjectInfoAction();
+            }
+            catch
+            {
+                if (failedActionReads++ > 0)
+                {
+                    throw;
+                }
+
+                Logger.LogError("Failed to read ProjectInfoAction from stream");
```

Comment:
Should this log the exception? I have no idea why this would fail, and need to be re-tried, but seems like the logs could help tell us.

Reply:
tbh it could be right now the only way it can throw is by getting an `Assumes.Unreachable` which isn't very actionable.

Reply:
In a perfect world, `Assumed.Unreachable` should never be hit. It should just be used to make the compiler happy. If it's being hit, then expectations are incorrect.

Reply:
I agree with that for sure, but trying to figure out why the expectations are incorrect is currently the hard part.